src/lib.rs

```diff
@@ -1035,6 +1054,84 @@ impl GetOptions {
         }
         Ok(())
     }
+
+    /// Create a new [`GetOptions`]
+    #[must_use]
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Sets the `if_match` condition.
+    ///
+    /// See [`GetOptions::if_match`]
+    #[must_use]
+    pub fn with_if_match(mut self, etag: impl Into<String>) -> Self {
```

Comment:
With this API there is no way to unset the `if_match` field (set it back to `None`)

I think it is more typical in builders to pass in the same type, something like

```suggestion
    pub fn with_if_match(mut self, etag: Option<impl Into<String>>) -> Self {
```

That would also reduce the number of API functions required by a factor of 2x

Reply:
I'm not sure I agree. Most of the builders I have seen either don't provide a way to supply an option and `with_x` has the assumption you're setting a value, or they provide a secondary `set_x` to set the specific `Option<T>`. For example:

* [`opentelemetry_sdk::metrics::MeterProviderBuilder`](https://docs.rs/opentelemetry_sdk/0.31.0/src/opentelemetry_sdk/metrics/meter_provider.rs.html#250) provides a `with_resource(mut self, resource: Resource)` where the underlying property is a `Option<Resource>`. It provides no alternative to set the option back to `None`
* [`std::sys::unix::process::Command`](https://stdrs.dev/nightly/x86_64-unknown-linux-gnu/src/std/sys/unix/process/process_common.rs.html#248) stores a `cwd: Option<CString>` but only provides `cwd(&mut self, dir: &OsStr)` with no alternative to set the option back to `None`

I'm not sure setting back to `None` is a common scenario, you should just build a new option?

Reply:
Actually, I thought about this some more and I think setting as an option does make more sense mainly because if you're setting an etag from a prior request (like the use case for https://github.com/apache/datafusion/issues/18211, or use cases that list files and then retrieve them) the data type in `ObjectMeta` is already an option. The chain becomes simpler then, because you don't have to unwrap the option and can just clone it directly into the builder.